app/controllers/bookmarks_controller.rb

```diff
@@ -61,7 +61,9 @@ def search
   def index
     if @bookmarkable
       access_denied unless is_admin? || @bookmarkable.visible?
-      @bookmarks = @bookmarkable.bookmarks.is_public.paginate(page: params[:page], per_page: ArchiveConfig.ITEMS_PER_PAGE)
+      @bookmarks = @bookmarkable.bookmarks.is_public
+      @bookmarks += @bookmarkable.bookmarks.where(hidden_by_admin: true) if is_admin?
```

Comment:
I think this is a bit weird how I'm doing it, but I didn't see another way to include bookmarks that have been hidden by admin and the public bookmarks. 

Also should I add check for `policy_and_abuse` role?

Reply:
Just a quick drive-by comment, not a thorough review:

1. Let's not check the admin role, since I'm pretty sure we don't currently check it for, e.g., accessing hidden works directly. But maybe we should changes this two instances of `is_admin?` to `logged_in_as_admin?` -- this seems to be the only file where we use `is_admin?` and that's weird, imo.
2. I'm guessing the problem with simply removing the `is_public` scope is you get private bookmarks as well as those hidden by admin? Would changing it to the `not_private` scope fix that? If not, I'd suggest defining a new scope in the bookmark model that will get public bookmarks, regardless of whether they're hidden by admin.

Reply:
1. I updated those `is_admin?` references to be `logged_in_as_admin?`
2. On a work's bookmarks page, users should only see public bookmarks, unless you're an admin, then they'll want to see public and hidden, right? Or should admins also be able to see private bookmarks on a work? Also, my changes here appear to work in that non-admins can only see public bookmarks and admins can see public and hidden; do you still think we should create a new scope? It's possible I'm misunderstanding the ask

Thank you!

Reply:
> On a work's bookmarks page, users should only see public bookmarks, unless you're an admin, then they'll want to see public and hidden, right?

Yes and the current code does that.

The way I understand sarken's suggestion is to use the `not_private` scope instead of `is_public`, since it gets the public bookmarks regardless of the hidden_by_admin status. Then filter out the hidden bookmarks depending on `logged_in_as_admin?`. That way, there aren't two queries that deal with `hidden_by_admin` (the `is_public` scope and the extra query with the `where` added in this PR), but only the one query (`not_private` and conditional `where` for hidden_by_admin on the returned relation).